execution/gethexec/caff_node.go

```diff
@@ -0,0 +1,204 @@
+package gethexec
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	espressoClient "github.com/EspressoSystems/espresso-sequencer-go/client"
+
+	"github.com/ethereum/go-ethereum/arbitrum_types"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/offchainlabs/nitro/arbos"
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+var (
+	retryTime = time.Second * 1
+)
+
+/*
+Caff Node creates blocks with finalized hotshot transactions
+*/
+type CaffNode struct {
+	stopwaiter.StopWaiter
+
+	config                  SequencerConfigFetcher
+	namespace               uint64
+	executionEngine         *ExecutionEngine
+	espressoClient          *espressoClient.Client
+	nextSeqBlockNum         uint64
```

Comment:
I know this param is what we did before. In the stateless batch poster design, should `nextSeqBlockNum` be got from seq inbox contract or from the batch?